c-dependencies/js-compute-runtime/builtins/backend.cpp

```diff
@@ -1,22 +1,606 @@
+  static constexpr auto kEECDH = "kEECDH";
+  /**
+   * Cipher suites using ephemeral ECDH key agreement, excluding anonymous cipher suites.
+   * Same as "kEECDH:-AECDH"
+   */
+  static constexpr auto EECDH = "EECDH";
+  /**
+   * Cipher suitesusing ECDH key exchange, including anonymous, ephemeral and fixed ECDH.
+   */
+  static constexpr auto ECDH = "ECDH";
+  /**
+   * Cipher suites using ephemeral ECDH key agreement, including anonymous cipher suites.
+   */
+  static constexpr auto kECDHE = "kECDHE";
+  /**
+   * Cipher suites using authenticated ephemeral ECDH key agreement
+   */
+  static constexpr auto ECDHE = "ECDHE";
+  /**
+   * Cipher suites using 128 bit AES.
+   */
+  static constexpr auto AES128 = "AES128";
+  /**
+   * Cipher suites using 256 bit AES.
+   */
+  static constexpr auto AES256 = "AES256";
+  /**
+   * Cipher suites using either 128 or 256 bit AES.
+   */
+  static constexpr auto AES = "AES";
+  /**
+   * AES in Galois Counter Mode (GCM): these cipher suites are only supported in TLS v1.2.
+   */
+  static constexpr auto AESGCM = "AESGCM";
+  /**
+   * Cipher suites using CHACHA20.
+   */
+  static constexpr auto CHACHA20 = "CHACHA20";
+  /**
+   * Cipher suites using triple DES.
+   */
+  static constexpr auto TRIPLE_DES = "3DES";
+  /**
+   * Cipher suites using SHA1.
+   */
+  static constexpr auto SHA1 = "SHA1";
+  /**
+   * Cipher suites using SHA1.
+   */
+  static constexpr auto SHA = "SHA";
+  /**
+   * Cipher suites using SHA256.
+   */
+  static constexpr auto SHA256 = "SHA256";
+  /**
+   * Cipher suites using SHA384.
+   */
+  static constexpr auto SHA384 = "SHA384";
+  // The content of the default list is determined at compile time and normally corresponds to
+  // ALL:!COMPLEMENTOFDEFAULT:!eNULL.
+  static constexpr auto DEFAULT = "DEFAULT";
+  // The ciphers included in ALL, but not enabled by default.
+  static constexpr auto COMPLEMENTOFDEFAULT = "COMPLEMENTOFDEFAULT";
+  static constexpr auto ALL = "ALL";
+
+  void moveToEnd(const AliasMap &aliases, std::vector<Cipher> &ciphers,
+                 std::string_view cipher) const {
+    this->moveToEnd(ciphers, aliases.at(cipher));
+  }
+
+  void moveToEnd(std::vector<Cipher> &ciphers,
+                 const std::vector<Cipher> &ciphersToMoveToEnd) const {
+    std::stable_partition(ciphers.begin(), ciphers.end(), [ciphersToMoveToEnd](auto cipher) {
+      return std::find(ciphersToMoveToEnd.begin(), ciphersToMoveToEnd.end(), cipher) ==
+             ciphersToMoveToEnd.end();
+    });
+  }
+
+  void add(const AliasMap &aliases, std::vector<Cipher> &ciphers, std::string_view alias) const {
+    auto toAdd = aliases.at(alias);
+    ciphers.insert(ciphers.end(), toAdd.begin(), toAdd.end());
+  }
+
+  void remove(const AliasMap &aliases, std::vector<Cipher> &ciphers, std::string_view alias) const {
+    auto &toRemove = aliases.at(alias);
+    ciphers.erase(std::remove_if(ciphers.begin(), ciphers.end(),
+                                 [&](auto x) {
```

Comment:
Referencing to avoid copying the cipher argument.
```suggestion
                                 [&](auto &x) {
```